### PR TITLE
[WIP] Event catalog

### LINKED ISSUE #1193: Create catalog of built-in events
Would there be interest in this? 

It seems like something similar to http://backbonejs.org/#Events-catalog could be useful.

### PR DESCRIPTION
**Pull request by mgrauer**
Fixes #1193.

This isn't done yet, but it's a lot of work, so I put an intermediate commit up now.  I'll continue on in this direction until I hear otherwise.

### CODE REVIEW

**Comment by brianhelba** [comment]
I'm renaming the PR to add a ``[WIP]`` prefix.

**Comment by mgrauer** [comment]
Thanks @brianhelba , I didn't put it on initially because I'd prefer feedback earlier.

**Comment by brianhelba** [comment]
@mgrauer Ah, sorry. I'll let you control the naming.

**Comment by mgrauer** [comment]
NP, it's kind of weird case.

**Comment by danlamanna** [comment]
@mgrauer Thanks for doing this, I was actually just thinking of taking a stab at it so I'm glad I checked first :P.

Does now seem like a good time to add very rudimentary jsdoc tags? (re @jbeezley in #420)

Since it will be a lengthy process, simply adding [@fires](http://usejsdoc.org/tags-fires.html) annotations might be a decent start to building up our JS documentation. I can go through and add them in this PR if it seems appropriate.

**Comment by mgrauer** [comment]
Sounds good to me @danlamanna .  I should be pushing another commit in the next hour.

**Comment by mgrauer** [comment]
Ready for review, or handoff to @danlamanna

File: clients/web/src/views/widgets/MetadataWidget.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ girder.views.MetadataWidget = girder.View.extend({
     initialize: function (settings) {
         this.item = settings.item;
         this.accessLevel = settings.accessLevel;
-        this.item.on('g:changed', function () {
+        this.item.on('g:fetched', function () {
```
Comment (zachmullen) [inline clients/web/src/views/widgets/MetadataWidget.js]: Why this change?

**Comment by zachmullen** [comment]
I think we need to organize these by what prototype/object triggers the event. This is mostly a documentation of certain aspects of various prototypes, and in the special case of `girder.events`, a centralized object for global events.

File: clients/web/src/views/widgets/MetadataWidget.js (hunk 1 above)
Comment (mgrauer) [inline clients/web/src/views/widgets/MetadataWidget.js]: Those are item models, which as far as I can tell, never trigger `g:changed` (collections do though).  Models trigger `g:fetched`, and that seems to be what those listeners are intended to do.

**Comment by jeffbaumes** [comment]
Much better than nothing - good start! Although I agree @zachmullen they need to be organized by the object that creates the event. It would make sense for these to be attached to the jsdoc for each type of object.

**Comment by jbeezley** [comment]
This is where having jsdoc infrastructure in place would be helpful.  If you add the [`@fires`](http://usejsdoc.org/tags-fires.html) annotation to the classes, you will get that grouping automatically in the documentation for each class.

**Comment by zachmullen** [comment]
Yep, that's what I would really love to have.

**Comment by mgrauer** [comment]
@danlamanna I'm not going to have any more time on this--would you want to take it over?

**Comment by danlamanna** [comment]
@mgrauer Yup I'll take a look

**Comment by zachmullen** [comment]
@mgrauer @danlamanna mind if I close this PR for cleanliness (unless you're actively working on it)? I'll leave the branch around.

**Comment by mgrauer** [comment]
no problem.  closed.
